python/grass/jupyter/interactivemap.py

```diff
@@ -347,13 +354,104 @@ def add_layer_control(self, **kwargs):
         else:
             self.layer_control_object = self._ipyleaflet.LayersControl(**kwargs)
 
+    def add_query_button(self):
+        """Add custom features like query button and coordinate retrieval"""
+        import ipywidgets as widgets
+        from IPython.display import HTML
+
+        # A button to activate/deactivate query mode
+        query_button = widgets.Button(description="Activate Query Mode")
+
+        # Variable to store the state of query mode
+        self.query_mode = False
+
+        # Output widget to display query results
+        output_widget = widgets.Output(
+            layout={
+                "display": "none",
+                "width": "100%",
+                "max_height": "300px",
+                "max_width": "300px",
+                "overflow": "auto",
+            }
+        )
+
+        # Function to toggle query mode
+        def toggle_query_mode(button):
+            self.query_mode = not self.query_mode
+            button.description = (
+                "Deactivate Query Mode" if self.query_mode else "Activate Query Mode"
+            )
+            output_widget.layout.display = "block" if self.query_mode else "none"
+            # Change cursor style based on query mode
+            if self.query_mode:
+                self.map.default_style = {"cursor": "crosshair"}
+            else:
+                self.map.default_style = {"cursor": "default"}
+
+        query_button.on_click(toggle_query_mode)
+
+        # Add the button to the map using WidgetControl
+        query_control = self._ipyleaflet.WidgetControl(
+            widget=query_button, position="topright"
+        )
+        self.map.add_control(query_control)
+
+        # Create an output widget to display query results
```

Comment:
Could the output be shown in a popup instead?

Reply:
I agree a pop up would be better here. Also, we could improve how the results are displayed. You could try a table with key, value pairs and the layer name as a header.